### PR TITLE
Kkraune/notebooks deploy MERGEOK

### PR DESCRIPTION
**Pull request by kkraune**
- Remove usage of default ports and disk folder in notebooks - easier to read/run with less noise
- Simplified getting started, moved the export/deploy to deploy-docker.ipynb where it belongs - will go over this and the other notebooks 
- bugfixed models in zipped package
- updated integration tests to not test with disk folder / _application_ folder inside app packages (support for the latter goes away in Vespa 8)
- no need for having hosts.xml, removed

I did not yet remove support for disk folder in deployment.py, the PR is large enough as it is - will remove in subsequent PR

### CODE REVIEW

File: vespa/test_integration_docker.py [hunk 1]
```diff
@@ -1030,91 +985,63 @@ def bert_model_input_and_output(
...
     def test_deploy_image(self):
         self.deploy(application_package=self.app_package,
-                    disk_folder=self.disk_folder,
                     container_image="vespaengine/vespa:7.566.21")
         self.vespa_docker.container.stop()
         self.vespa_docker.container.remove()
 
     def tearDown(self) -> None:
-        shutil.rmtree(self.disk_folder, ignore_errors=True)
+        pass
```
Comment (thigm85) [inline vespa/test_integration_docker.py:1038]: Should we remove the tearDown method if it is not doing anything?

**Review by thigm85 — CHANGES_REQUESTED** [review]

File: vespa/test_integration_docker.py [hunk 2]
```diff
@@ -132,70 +132,25 @@ def create_sequence_classification_task():
...
-    def deploy_from_disk_with_application_folder(
-        self, application_package, disk_folder
-    ):
-        self.vespa_docker = VespaDocker(port=8089, disk_folder=disk_folder)
-        self.vespa_docker.export_application_package(
-            application_package=application_package
-        )
         #
-        # Application folder inside disk folder
+        # Test VespaDocker serialization
```
[the remark this answers is not in the document]
Comment (kkraune) [reply vespa/test_integration_docker.py:146]: I need to work more on the serialization test, so left it as a todo for myself. but can keep it and fix if there are issues

File: vespa/test_integration_docker.py [hunk 3]
```diff
@@ -132,70 +132,25 @@ def create_sequence_classification_task():
...
+            self.vespa_docker = VespaDocker(port=8089, container_image=container_image)
         else:
-            self.vespa_docker = VespaDocker(port=8089, disk_folder=disk_folder)
+            self.vespa_docker = VespaDocker(port=8089)
 
-        app = self.vespa_docker.deploy(application_package=application_package)
-        #
-        # Test deployment
-        #
-        self.assertTrue(
```
[the remark this answers is not in the document]
Comment (kkraune) [reply vespa/test_integration_docker.py:145]: will look into

File: vespa/test_integration_docker.py [hunk 4]
```diff
@@ -132,70 +132,25 @@ def create_sequence_classification_task():
...
-        )
-
-    def deploy_without_disk_folder(self, application_package):
-        self.vespa_docker = VespaDocker(port=8089)
         try:
             app = self.vespa_docker.deploy(application_package=application_package)
         except RuntimeError as e:
             assert False, "Deployment error: {}".format(e)
-
-    def deploy_from_disk_with_disk_folder(self, application_package, disk_folder):
```
[the remark this answers is not in the document]
Comment (kkraune) [reply vespa/test_integration_docker.py:163]: I don't think we should keep deploying from disk. It complicates as you must link the disk into docker - we have removed this from all other vespa doc / samples. We want users to always deploy using a zip, as the migration to services like Vespa Cloud is then easier, as it is essentially the same. As we must have the ziop deployment anyway for cloud, it is less code to have only that as an option. I might have overlooked benefits of deploying from disk, are there any?

File: vespa/test_integration_docker.py (hunk 1 above)
Comment (kkraune) [reply vespa/test_integration_docker.py:1038]: I considered that, I kept if for symmetry with the others, can remove

File: vespa/test_integration_docker.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (thigm85) [reply vespa/test_integration_docker.py:163]: Benefits of deploying from disk happen when the users want to change application package files manually, e.g. when making changes not supported by pyvespa API.

Is this use case still supported?

[the remark this answers is not in the document]
Comment (kkraune) [reply vespa/test_integration_docker.py:163]: yes, you can still export, change and deploy. the deploy will then zip the changed files. so no change for the user

[the remark this answers is not in the document]
Comment (thigm85) [reply vespa/test_integration_docker.py:163]: Great 👍

File: vespa/test_integration_docker.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (kkraune) [reply vespa/test_integration_docker.py:145]: as we deploy using POST, the _Generation ..._ output is not available. `deploy` will fail with a runtime error if status code != 200, and the test will then fail

File: vespa/test_integration_docker.py (hunk 1 above)
Comment (kkraune) [reply vespa/test_integration_docker.py]: removed

**Review by thigm85 — APPROVED** [review]

**Pull request merged**